src/components/AccountSettings.tsx

```diff
@@ -1,330 +1,277 @@
-import { getSourceAppRedirectURL } from './SourceApp'
-import { ConfigureEmail } from './ConfigureEmail'
-import { UnbindORCiDDialog } from './UnbindORCiD'
-import { ORCiDButton } from './ORCiDButton'
+import IconSvg from 'synapse-react-client/dist/containers/IconSvg'
 import { useHistory } from 'react-router-dom'
+import { ORCiDButton } from './ORCiDButton'
 
-export type AccountSettingsProps = {
-}
-
-const AccountSettings = (props: AccountSettingsProps) => {
-    const { accessToken } = useSynapseContext()
-    const [ userProfile, setUserProfile ] = useState<UserProfile>()
-    const [ showDialog, setShowDialog ] = useState<boolean>(false)
-    const [ orcid, setOrcid ] = useState<string>()
-    const [ verified, setVerfied ] = useState<boolean>()
-    const [ isCertified, setIsCertified] = useState<boolean>()
-    const [ editUsername, setEditUsername] = useState<boolean>(false)
-    const [ changePW, setChangePW] = useState<boolean>(false)
-    const [ isShowingWelcomeScreen, setIsShowingWelcomeScreen] = useState<boolean>(false)
-    const [ updatedUsername, setUpdatedUsername] = useState<string>('')
-    const [ verificationSubmission, setVerificationSubmission] = useState<VerificationSubmission>()
-    const [ verificationState, setVerificationState] = useState<VerificationState>()
-    const [ showORCiDDialog, setShowORCiDDialog ] = useState(false)
-
-    const SUSPENDED_TEXT = 'Your account has been suspended.'
-    const REJECTED_TEXT = 'Sorry we could not verify your account.'
+export const AccountSettings = () => {
+  const { accessToken } = useSynapseContext()
+  const [ userProfile, setUserProfile ] = useState<UserProfile>()
+  const [ firstName, setFirstName ] = useState<string>()
+  const [ lastName, setLastName ] = useState<string>()
+  const [ position, setPosition ] = useState<string>()
+  const [ company, setCompany ] = useState<string>()
+  const [ location, setLocation ] = useState<string >()
+  const [ url, setUrl ] = useState<string>()
+  const [ industry, setIndustry ] = useState<string>()
+  const [ primaryEmail, setPrimaryEmail ] = useState<string>()
+  const [ username, setUsername ] = useState<string>()
+  const [ changeInForm, setChangeInForm] = useState(false)
+  const [ orcid, setOrcid ] = useState<string>()
+  const [ verified, setVerified ] = useState<boolean>()
+  const [ isCertified, setIsCertified ] = useState<boolean>()
+  const history = useHistory()
 
-    const history = useHistory()
+  const profileInformationRef = useRef<HTMLDivElement>(null)
+  const changePasswordRef = useRef<HTMLDivElement>(null)
+  const trustCredentialRef= useRef<HTMLDivElement>(null)
+  const personalAccessTokenRef = useRef<HTMLDivElement>(null)
 
-    const handleChangesFn = (val: string) => {
-        history.push(`/authenticated/${val}`)
-    }
+  const handleChangesFn = (val: string) => {
+    history.push(`/authenticated/${val}`)
+  }
+  
+  const markFormDirty = () => setChangeInForm(true)
```

Comment:
When any sort of text is written in the form below this changes the disabled button. However, when the changed value is reverted back the button is still enabled.
There needs to be a way to disable the button when value has not changed.